streaming/stream_transfer_task.cc

```diff
@@ -148,13 +147,15 @@ future<> send_mutation_fragments(lw_shared_ptr<send_info> si) {
             });
         }();
 
-        auto sink_op = [sink, si, got_error_from_peer] () mutable -> future<> {
+        auto sink_op = [sink, si, got_error_from_peer, table_is_dropped] () mutable -> future<> {
             mutation_fragment_stream_validator validator(*(si->reader.schema()));
-            return do_with(std::move(sink), std::move(validator), [si, got_error_from_peer] (rpc::sink<frozen_mutation_fragment, stream_mutation_fragments_cmd>& sink, mutation_fragment_stream_validator& validator) {
-                return repeat([&sink, &validator, si, got_error_from_peer] () mutable {
-                    return si->reader().then([&sink, &validator, si, s = si->reader.schema(), got_error_from_peer] (mutation_fragment_opt mf) mutable {
+            return do_with(std::move(sink), std::move(validator), [si, got_error_from_peer, table_is_dropped] (rpc::sink<frozen_mutation_fragment, stream_mutation_fragments_cmd>& sink, mutation_fragment_stream_validator& validator) {
+                return repeat([&sink, &validator, si, got_error_from_peer, table_is_dropped] () mutable {
+                    return si->reader().then([&sink, &validator, si, s = si->reader.schema(), got_error_from_peer, table_is_dropped] (mutation_fragment_opt mf) mutable {
                         if (*got_error_from_peer) {
                             return make_exception_future<stop_iteration>(std::runtime_error("Got status error code from peer"));
+                        } else if (*table_is_dropped) {
+                            return make_ready_future<stop_iteration>(stop_iteration::yes);
```

Comment:
This will make the sender sending stream_mutation_fragments_cmd::end_of_stream which means the stream has officially finished successfully which is not true.  In case the table is dropped, we'd better skip sending end_of_stream.

On the receiver,  we need to adjust the handling a bit, we can extend stream_mutation_fragments_cmd_status to have a flag to record if the table is dropped, if the table is dropped, do not check end_of_stream is received. 
```
                            // If the sender has sent stream_mutation_fragments_cmd it means it is
                            // a node that understands the new protocol. It must send end_of_stream
                            // before close the stream.
                            if (cmd_status->got_cmd && !cmd_status->got_end_of_stream) {
                                return make_exception_future<mutation_fragment_opt>(std::runtime_error("Sender did not sent end_of_stream"));
                            }
```